src/posix/platform/radio.cpp

```diff
@@ -94,18 +94,22 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
     SuccessOrDie(sRadioSpinel.SetPromiscuous(aEnable));
 }
 
-void platformRadioInit(const otPlatformConfig *aPlatformConfig)
+void platformRadioInit(otPosixRadioArguments *aArguments)
 {
+    ot::Posix::Arguments *args           = reinterpret_cast<ot::Posix::Arguments *>(aArguments);
+    bool                  resetRadio     = (args->GetValue("no-reset") == NULL);
+    bool                  restoreDataset = (args->GetValue("restore-dataset") != NULL);
```

Comment:
use the original parameter name?
```suggestion
    bool                  restoreDataset = (args->GetValue("ncp-dataset") != NULL);
```